metal3-io/cluster-api-provider-metal3
# Add cert-rotation e2e test

This PR adds an e2e test for the Ironic certificate rotation feature.
The Ironic certificate rotation feature introduces the use of cert-manger to generate new TLS certificates and renew these certificates when they are expired. The current integration test can ensure if ironic can adopt the new certificates generated by cert-manager, so this PR aims to test the if certificates can be renewed, and ironic can continue to function well with the new certificates.
Test scenario:
- Manually trigger the certificate renewal by deleting the secrets that contain Ironic certificates.
- Check if ironic contains which use the certificates can restart. After restarting, they will adopt the new certificates.
- Check if the containers are in the running state.
